test/e2e/epoching_spam_prevention_e2e_test.go

```diff
@@ -0,0 +1,557 @@
+package e2e
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/babylonlabs-io/babylon/v4/test/e2e/configurer"
+	"github.com/babylonlabs-io/babylon/v4/test/e2e/configurer/chain"
+	"github.com/babylonlabs-io/babylon/v4/test/e2e/util"
+	"github.com/babylonlabs-io/babylon/v4/testutil/datagen"
+	etypes "github.com/babylonlabs-io/babylon/v4/x/epoching/types"
+
+	sdked25519 "github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+)
+
+type EpochingSpamPreventionTestSuite struct {
+	suite.Suite
+
+	configurer configurer.Configurer
+}
+
+// waitForEpochEnd waits for the current epoch to end and processes queued messages
+func (s *EpochingSpamPreventionTestSuite) waitForEpochEnd(chainA *chain.Config, nonValidatorNode *chain.NodeConfig, stepName string) {
+	s.T().Logf("Step 2: %s - Waiting for epoch end to process enqueued message", stepName)
+
+	// Query current epoch information including epoch boundary
+	currentEpochResp, err := func() (*etypes.QueryCurrentEpochResponse, error) {
```

Comment:
we can use [nonValidatorNode.QueryCurrentEpoch](https://github.com/babylonlabs-io/babylon/blob/8cb9862b1d0631edd15df1039085afb4274823a9/test/e2e/configurer/chain/queries.go#L322)